### PR TITLE
python311Packages.color-operations: 0.1.3 -> 0.1.4

### PR DESCRIPTION
**Pull request by GaetanLepage**
## Description of changes

~~Fails on both `aarch64-darwin` and `aarch64-linux` with:~~

-> Update the package to its latest version to fix the ARM builds.

cc @NixOS/geospatial 

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [x] aarch64-linux
  - [x] x86_64-darwin
  - [x] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) (or backporting [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) and [24.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

**Comment by GaetanLepage** [comment]
Result of `nixpkgs-review pr 320527` run on aarch64-darwin [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>4 packages failed to build:</summary>
  <ul>
    <li>python311Packages.rio-tiler</li>
    <li>python311Packages.rio-tiler.dist</li>
    <li>python312Packages.rio-tiler</li>
    <li>python312Packages.rio-tiler.dist</li>
  </ul>
</details>
<details>
  <summary>4 packages built:</summary>
  <ul>
    <li>python311Packages.color-operations</li>
    <li>python311Packages.color-operations.dist</li>
    <li>python312Packages.color-operations</li>
    <li>python312Packages.color-operations.dist</li>
  </ul>
</details>

**Comment by GaetanLepage** [comment]
Result of `nixpkgs-review pr 320527` run on x86_64-darwin [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>8 packages built:</summary>
  <ul>
    <li>python311Packages.color-operations</li>
    <li>python311Packages.color-operations.dist</li>
    <li>python311Packages.rio-tiler</li>
    <li>python311Packages.rio-tiler.dist</li>
    <li>python312Packages.color-operations</li>
    <li>python312Packages.color-operations.dist</li>
    <li>python312Packages.rio-tiler</li>
    <li>python312Packages.rio-tiler.dist</li>
  </ul>
</details>

**Comment by GaetanLepage** [comment]
Result of `nixpkgs-review pr 320527` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>8 packages built:</summary>
  <ul>
    <li>python311Packages.color-operations</li>
    <li>python311Packages.color-operations.dist</li>
    <li>python311Packages.rio-tiler</li>
    <li>python311Packages.rio-tiler.dist</li>
    <li>python312Packages.color-operations</li>
    <li>python312Packages.color-operations.dist</li>
    <li>python312Packages.rio-tiler</li>
    <li>python312Packages.rio-tiler.dist</li>
  </ul>
</details>

**Comment by GaetanLepage** [comment]
Result of `nixpkgs-review pr 320527` run on aarch64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>4 packages failed to build:</summary>
  <ul>
    <li>python311Packages.rio-tiler</li>
    <li>python311Packages.rio-tiler.dist</li>
    <li>python312Packages.rio-tiler</li>
    <li>python312Packages.rio-tiler.dist</li>
  </ul>
</details>
<details>
  <summary>4 packages built:</summary>
  <ul>
    <li>python311Packages.color-operations</li>
    <li>python311Packages.color-operations.dist</li>
    <li>python312Packages.color-operations</li>
    <li>python312Packages.color-operations.dist</li>
  </ul>
</details>

File: pkgs/development/python-modules/color-operations/default.nix [hunk 1]
```diff
@@ -42,6 +42,11 @@ buildPythonPackage rec {
 
   pythonImportsCheck = [ "color_operations" ];
 
+  disabledTests = lib.optionals stdenv.isAarch64 [
+    # AsserionError: assert -2.220446049250313e-16 >= 0
+    "test_rgb2lch"
```
Comment (l0b0) [inline pkgs/development/python-modules/color-operations/default.nix:47]: Would you be OK to [report this upstream](https://github.com/vincentsarago/color-operations/issues/new)?

File: pkgs/development/python-modules/color-operations/default.nix [hunk 2]
```diff
@@ -1,14 +1,14 @@
...
-
-  colormath,
+  fetchFromGitHub,
   cython,
-  oldest-supported-numpy,
   setuptools,
+  oldest-supported-numpy,
+  colormath,
+  pytestCheckHook,
+  stdenv,
```
Comment (l0b0) [inline pkgs/development/python-modules/color-operations/default.nix:11]: Why reorder these? Especially away from a grouped + sorted order? The new order looks arbitrary.

Comment (GaetanLepage) [reply pkgs/development/python-modules/color-operations/default.nix:11]: The order is the one recommended in the manual, i.e. the order of apparition in the derivation.
I added comments to highlight the categories.

File: pkgs/development/python-modules/color-operations/default.nix (hunk 1 above)
Comment (GaetanLepage) [reply pkgs/development/python-modules/color-operations/default.nix:47]: Sure ! Done: https://github.com/vincentsarago/color-operations/issues/9

Comment (GaetanLepage) [reply pkgs/development/python-modules/color-operations/default.nix]: The test has been fixed upstream.

**Comment by GaetanLepage** [comment]
Result of `nixpkgs-review pr 320527` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>8 packages built:</summary>
  <ul>
    <li>python311Packages.color-operations</li>
    <li>python311Packages.color-operations.dist</li>
    <li>python311Packages.rio-tiler</li>
    <li>python311Packages.rio-tiler.dist</li>
    <li>python312Packages.color-operations</li>
    <li>python312Packages.color-operations.dist</li>
    <li>python312Packages.rio-tiler</li>
    <li>python312Packages.rio-tiler.dist</li>
  </ul>
</details>

---
Result of `nixpkgs-review pr 320527` run on aarch64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>4 packages failed to build:</summary>
  <ul>
    <li>python311Packages.rio-tiler</li>
    <li>python311Packages.rio-tiler.dist</li>
    <li>python312Packages.rio-tiler</li>
    <li>python312Packages.rio-tiler.dist</li>
  </ul>
</details>
<details>
  <summary>4 packages built:</summary>
  <ul>
    <li>python311Packages.color-operations</li>
    <li>python311Packages.color-operations.dist</li>
    <li>python312Packages.color-operations</li>
    <li>python312Packages.color-operations.dist</li>
  </ul>
</details>

---

Result of `nixpkgs-review pr 320527` run on aarch64-darwin [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>4 packages failed to build:</summary>
  <ul>
    <li>python311Packages.rio-tiler</li>
    <li>python311Packages.rio-tiler.dist</li>
    <li>python312Packages.rio-tiler</li>
    <li>python312Packages.rio-tiler.dist</li>
  </ul>
</details>
<details>
  <summary>4 packages built:</summary>
  <ul>
    <li>python311Packages.color-operations</li>
    <li>python311Packages.color-operations.dist</li>
    <li>python312Packages.color-operations</li>
    <li>python312Packages.color-operations.dist</li>
  </ul>
</details>

---

Result of `nixpkgs-review pr 320527` run on x86_64-darwin [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>8 packages built:</summary>
  <ul>
    <li>python311Packages.color-operations</li>
    <li>python311Packages.color-operations.dist</li>
    <li>python311Packages.rio-tiler</li>
    <li>python311Packages.rio-tiler.dist</li>
    <li>python312Packages.color-operations</li>
    <li>python312Packages.color-operations.dist</li>
    <li>python312Packages.rio-tiler</li>
    <li>python312Packages.rio-tiler.dist</li>
  </ul>
</details>

File: pkgs/development/python-modules/color-operations/default.nix [hunk 3]
```diff
@@ -1,35 +1,40 @@
...
     owner = "vincentsarago";
     repo = "color-operations";
-    rev = version;
-    hash = "sha256-KsrgilcNK2ufPKrhtGdf8mdlFzhsHB2jHN+WDlZqabc=";
+    rev = "refs/tags/${version}";
+    hash = "sha256-qqOTmVYD3VfjeVJtYvDQw+cxjcTsmqTYQNL1qMX+fL4=";
   };
 
-  nativeBuildInputs = [
+  build-system = [
```
Comment (natsukium) [inline pkgs/development/python-modules/color-operations/default.nix:32]:
```suggestion
  build-system = [
    oldest-supported-numpy
```
Could you please add `oldest-supported-numpy` to `build-system`?
https://github.com/vincentsarago/color-operations/blob/d76a3464bef29e7c2ac29534f474f1f5229237d1/pyproject.toml#L1-L7

File: pkgs/development/python-modules/color-operations/default.nix [hunk 4]
```diff
@@ -1,35 +1,40 @@
...
     cython
     setuptools
   ];
 
-  propagatedBuildInputs = [ oldest-supported-numpy ];
+  dependencies = [ oldest-supported-numpy ];
```
Comment (natsukium) [inline pkgs/development/python-modules/color-operations/default.nix:37]:
```suggestion
  dependencies = [ numpy ];
```
https://github.com/vincentsarago/color-operations/blob/d76a3464bef29e7c2ac29534f474f1f5229237d1/setup.py#L94

**Review by natsukium — APPROVED** [review]
Thanks.

**Pull request merged**